runtime/criusupport/criusupport.c

```diff
@@ -298,49 +329,46 @@ Java_org_eclipse_openj9_criu_CRIUSupport_checkpointJVMImpl(JNIEnv *env,
 		vmFuncs->acquireExclusiveVMAccess(currentThread);
 
 		if (FALSE == vmFuncs->jvmCheckpointHooks(currentThread)) {
-			resultType = vm->criuSupportJVMCheckpointFailure;
 			goto releaseExclusive;
 		}
 
 		systemReturnCode = criu_dump();
 		if (systemReturnCode < 0) {
-			vmFuncs->setCurrentExceptionNLSWithArgs(currentThread, J9NLS_JCL_CRIU_DUMP_FAILED, J9VMCONSTANTPOOL_JAVALANGINTERNALERROR, systemReturnCode);
-			resultType = vm->criuSupportSystemCheckpointFailure;
+			currentExceptionClass = vm->criuSystemCheckpointExceptionClass;
+			nlsMsgFormat = j9nls_lookup_message(J9NLS_DO_NOT_PRINT_MESSAGE_TAG | J9NLS_DO_NOT_APPEND_NEWLINE, J9NLS_JCL_CRIU_DUMP_FAILED, NULL);
 			goto releaseExclusive;
 		}
 
 		/* We can only end up here if the CRIU restore was successful */
 		isAfterCheckpoint = TRUE;
 
 		if (FALSE == vmFuncs->jvmRestoreHooks(currentThread)) {
-			resultType = vm->criuSupportJVMRestoreFailure;
 			goto releaseExclusive;
 		}
```

Comment:
@DanHeidinga Is it fine if I leave the exception set by the hook?

Reply:
It's OK for now as this is an experimental API.  We may want to change it once we have more experience with writing hooks so that we wrap it with a RestoreException but that can be looked at in the future.  It really depends on how clear and debugable the solution ends up being and we won't know that till we play with it